out-adapter-persistence-jpa/src/test/java/gohigher/application/ApplicationPersistenceQueryAdapterTest.java

```diff
@@ -0,0 +1,81 @@
+package gohigher.application;
+
+import static gohigher.fixture.ApplicationFixture.*;
+import static gohigher.fixture.ProcessFixture.*;
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.*;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import gohigher.application.entity.ApplicationJpaEntity;
+import gohigher.application.entity.ApplicationRepository;
+
+@DisplayName("ApplicationPersistenceQueryAdapter 클래스의")
+@ExtendWith(MockitoExtension.class)
+class ApplicationPersistenceQueryAdapterTest {
+
+	@Mock
+	private ApplicationRepository applicationRepository;
+
+	private ApplicationPersistenceQueryAdapter applicationPersistenceQueryAdapter;
+
+	@BeforeEach
+	void setUp() {
+		applicationPersistenceQueryAdapter = new ApplicationPersistenceQueryAdapter(applicationRepository);
+	}
+
+	@DisplayName("findByIdAndMonth 메서드는")
+	@Nested
+	class Describe_findByIdAndMonth {
+
+		private final long userId = 1L;
+		private final int year = 2023;
+		private final int month = 9;
+
+		@DisplayName("해당 연/월에 유저가 등록한 지원서 일정이 정보가 있을 때")
+		@Nested
+		class Context_with_many_schedules_for_several_months {
+
+			@BeforeEach
+			void setUp() {
+				ApplicationJpaEntity naverApplication = NAVER_APPLICATION.toEntity(userId);
+				naverApplication.getProcesses()
+					.add(TEST.toApplicationProcessEntity(naverApplication, LocalDate.of(year, month, 20)));
+				naverApplication.getProcesses()
+					.add(TO_APPLY.toApplicationProcessEntity(naverApplication, LocalDate.of(year, month, 11)));
+
+				ApplicationJpaEntity kakaoApplication = KAKAO_APPLICATION.toEntity(userId);
+				kakaoApplication.getProcesses()
+					.add(TO_APPLY.toApplicationProcessEntity(kakaoApplication, LocalDate.of(year, month, 20)));
+
+				List<ApplicationJpaEntity> applicationJpaEntities = List.of(naverApplication, kakaoApplication);
+
+				given(applicationRepository.findByUserIdAndMonth(userId, year, month))
+					.willReturn(applicationJpaEntities);
+			}
+
+			@DisplayName("일정 정보가 담긴 지원서를 반환한다.")
+			@Test
+			void it_return_application_with_processes() {
+				List<Application> applications = applicationPersistenceQueryAdapter.findByIdAndMonth(userId, year,
+					month);
+
+				assertAll(
+					() -> assertThat(applications).hasSize(2),
```

Comment:
```suggestion
					() -> assertThat(applications).hasSize(applicationJpaEntities.size()),
```
저장 전 리스트의 길이로 받으면 테스트 코드 수정에도 용이하고 2의 의미도 명확히 드러낼 수 있을것 같습니다!

Reply:
그리고 다음에 테스트가 추가될 것을 고려해서 테스트 격리가 필요해보입니다!

Reply:
피드백을 반영하여 `2`와 같은 매직넘버들은 의미를 알 수 있게 수정했습니다!!
의미를 표현하기 힘든 경우는 주석을 추가했습니다!

Reply:
혹시 테스트 격리에 대해서는 어떤 것을 말씀하시는건지 여쭈어봐도 될까요...?
`RepositoryTest`의 경우 `@DataJpaTest`가 `@Transactional` 어노테이션을 갖고 있어서 테스트별로 실행될 때, 데이터베이스에 실제 저장되지 않고 Rollback 처리가 되어 격리가 되는 상태라 생각됩니다. 혹시 우려되는 상황이 있을까요?

Reply:
테스트 시 실제 데이터베이스를 사용하기 때문에, 로컬 데이터베이스에 데이터가 있는 상태에서 메서드를 호출하면 결과와 다르게 나올수도 있을것 같습니다! (저장되어 있던 데이터 + 테스트용으로 저장한 데이터)

Reply:
간단하게 생각한 해결책으로는 현재 문제가 발생할 것이 기존에 저장된 데이터가 테스트에 끼칠 영향을 없애는 것이기에 `@GeneratedValue`에서 사용하지 않는 id값(0, -1 등)을 통해 테스트를 하는 것이 떠올라 적용하였습니다.

Reply:
넵넵 TestEntityManager 로 테스트 격리가 가능하다면 사실 문제 없을거같긴 합니다! ㅎㅎ 직접 테스트 하고 싶었으나 제 코드와 충돌이 나서 확인이 어렵네요. 😅